### DISCUSSION TITLE
How about some javascript tests with jest ?

### DISCUSSION THREAD

**Discussion by TrimmingFool** [Novik/ruTorrent#2314] [Show and tell]
@Novik @stickz 

For sane refactoring you need some tests.
Even though ruTorrent is quite old and does not use ES modules, it is still possible to test some functionality in isolation, e.g.,  the xmlrpc response parsing in `js/rtorrent.js` with jest + jsdom + jquery.
I did this here to test the refactoring of `js/rtorrent.js` with `tests/js/rtorrent.spec.js`: 

TrimmingFool/ruTorrent/pull/1

Reference: https://github.com/facebook/jest/tree/main/examples/jquery https://jestjs.io/docs/tutorial-jquery

Nevertheless, I also see merit in the strategy of keeping things as they are and just avoiding refactoring. (since everything is so old)
(Also at the moment I personally do not have any further refactorings in mind)

**Comment by stickz** [comment]
I think we should go for it, if @Novik doesn't have any objections. Would you mind waiting a couple weeks though? I would like to get anther pre-release of ruTorrent created to transition more users to version 4. This has a very high potential to introduce regressions.

**Reply by TrimmingFool** [reply]
> Would you mind waiting a couple weeks though?

Sure, there is no hurry.

> I would like to get another pre-release of ruTorrent created to transition more users to version 4.

Sounds great :+1:

**Reply by stickz** [reply]
@TrimmingFool You're good to go with the code refactoring. Feel free to create a pull request, if you're willing to support these changes and resolve any regressions. You'll need to request a review from @Novik. I'm not an expert on this portion of the code.
